File: C#-part-2/03.Methods/05.Larger than neighbours/LargerThanNeighbours.cs
//Write program that reads an array of numbers and prints how many of them are larger than their neighbours.
namespace _05.Larger_than_neighbours
{
    using System;
    using System.Linq;

    class LargerThanNeighbours
    {
        static int FindLargerThanTwo(int[] arr)
        {
            int counter = 0;

            for (int i = 1; i < arr.Length-1; i++)
            {
                if (arr[i] >arr[i+1]&&arr[i]>arr[i-1])
                {
                    counter++;
                }
            }
              
            return counter;
        }

        static void Main()
        {
            int n = int.Parse(Console.ReadLine());
            var input = Console.ReadLine().Split(' ').Select(x => int.Parse(x)).ToArray();
            int result = FindLargerThanTwo(input);

            Console.WriteLine(result);
        }
    }
}

File: Homeworks/2. Web Design and UI technlogies/3. JavaScript-Part-1/06.Functions/05.Larger than neighbours/Larger than neighbours.js
/**  Problem 5.Larger than neighbours
 * Write a method that checks if the element at given position in given array of integers is larger than its two neighbours (when such exist).
 * Write program that reads an array of numbers and prints how many of them are larger than their neighbours.
 * Input               Output
 6                    2
 -26 -25 -28 31 2 27
 */
function largerThanNeighbours(args) {
    var arrayLength =args[0],
        array = args[1].split(' ').map(Number);

    return neighboursChecker(array, arrayLength);

    function neighboursChecker(sequence, arrLength) {
        var counter = 0,
            i;
        for (i = 1; i < arrLength - 1; i++) {
            if (sequence[i - 1] < sequence[i] && sequence[i + 1] < sequence[i]) {
                counter += 1;
            }
        }

        return counter;
    }
}
// Test
var res = largerThanNeighbours(['6', '-26 -25 -28 31 2 27']);
console.log(res);